docs/book/src/cronjob-tutorial/testdata/project/controllers/cronjob_controller_test.go

```diff
@@ -0,0 +1,200 @@
+/*
+We can create the tests in a separate file, cronjob_controller_test.go.
+Keeping consistent with the Gingko usage in our autogenerated suite_test.go code, we’ll use Gingko to write our tests.
+*/
+
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+// +kubebuilder:docs-gen:collapse=Apache License
+
+/*
+As usual, we start with necessary imports. We also define some utility variables.
+*/
+package controllers
+
+import (
+	"context"
+	"reflect"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	batchv1 "k8s.io/api/batch/v1"
+	batchv1beta1 "k8s.io/api/batch/v1beta1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+
+	cronjobv1 "tutorial.kubebuilder.io/project/api/v1"
+)
+
+const (
+	CronjobName      string = "test-cronjob"
+	CronjobNamespace string = "test-cronjob-namespace"
+	JobName          string = "test-job"
+
+	timeout  = time.Second * 10
+	interval = time.Millisecond * 250
+)
+
+/*
+We also expose the GVK for our CronJob version, which we will use to manage ownership of Jobs that belong to our test CronJob.
+One way you can set this GVK is by defining it as a global var like so:
+*/
+var (
+	Kind string                  = reflect.TypeOf(cronjobv1.CronJob{}).Name()
+	GVK  schema.GroupVersionKind = cronjobv1.GroupVersion.WithKind(Kind)
+)
+
+/*
+Now we set up the Gingko testing framework to run CronJob controller tests.
+
+The first step to writing a simple integration test is to actually create an instance of CronJob you can run tests against.
+Note that to create a CronJob, you’ll need to create a stub CronJob struct that contains your CronJob’s specifications.
+
+To create a stub CronJob, we'll also need to create stubs of its required downstream objects--namely, its Job template spec and
+its Pod template spec.
```

Comment:
The text here is not very clear for me. Also, is it something that  is vital for the users get how it works or is something that we could just remove?

Reply:
What I try to convey here is that to create a stubbed CronJob, you'll also have to create stubbed downstream objects like the `CronJobSpec` required by the CronJob's `Spec` field, the `JobSpec` required by the `CronJobSpec`, and the `PodSpecTempate` and `PodSpec` required by the `JobSpec`. I wanted to emphasize that a tester could not just create a `CronJob` object but would also have to be responsible for creating some downstream objects to initialize a CronJob object correctly.

I tried to clear up the phrasing a bit--let me know what you think!